content/client-lib-development-guide/features.textile

```diff
@@ -488,6 +490,11 @@ h3(#realtime-channel). Channel
 ** @(RTL4f)@ Once an @ATTACH@ @ProtocolMessage@ is sent, if an @ATTACHED@ @ProtocolMessage@ is not received within the "default realtime request timeout":#defaults, the attach request should be treated as though it has failed and the channel should transition to the @SUSPENDED@ state. The channel will then be subsequently automatically re-attached as described in "RTL13":#RTL13
 ** @(RTL4d)@ If the language permits, a callback can be provided that is called when the channel is attached successfully or the attach fails and the @ErrorInfo@ error is passed as an argument to the callback
 ** @(RTL4e)@ If the user does not have sufficient permissions to attach to the channel, the channel will transition to @FAILED@ and set the @Channel#errorReason@
+** @(RTL4j)@ If the attach is not a clean attach (where a 'clean attach' is an attach attempt where the channel has either not previously been attached or has been explicitly detached since the last time it was attached); for example an automatic reattach triggered by "@RTN15c3@":#RTN15c3 or "@RTL13a@":#RTL13a (non-exhaustive); the library should set the "@ATTACH_RESUME@":#TR3f flag in the @ATTACH@ message
```

Comment:
Can we provide a hint to a developer in regards to how they can test this behaviour with the realtime system?

Reply:
https://github.com/ably/docs/pull/766/files#diff-11900b395df266bc2bbfc1d11bdfc7a0R494